subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/ProjectExtensions.kt

```diff
@@ -106,7 +114,11 @@ inline fun <reified T : Any> Project.the(): T =
 
 /**
  * Returns the plugin convention or extension of the specified type.
+ *
+ * Note, that the concept of conventions is deprecated and scheduled for
+ * removal in Gradle 8.
  */
+@Suppress("deprecation")
 fun <T : Any> Project.the(extensionType: KClass<T>): T =
     convention.findByType(extensionType.java)
```

Comment:
💄 Move the suppression closer to the deprecated member usage
```suggestion
fun <T : Any> Project.the(extensionType: KClass<T>): T =
    @Suppress("deprecation")
    convention.findByType(extensionType.java)
```